chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt

```diff
@@ -660,5 +659,39 @@ class RoomSpek : Spek({
 
             assertThat(isSubscribed).isFalse()
         }
+
+        it("does not emit onUserJoined for members of a room when the subscription is first created") {
+
+            setUpInstanceWith(createDefaultRole(), newUsers(ALICE, PUSHERINO), newRoom(GENERAL, PUSHERINO, SUPER_USER))
+
+            var usersJoined = 0
+            var roomUpdated = CountDownLatch(1)
```

Comment:
I was confused by the fact that we replace the latch behind this variable part way through the test when I was reading it - perhaps we could create one variable for each latch and name them after the events to make it clearer when we wait for which event.

Reply:
I've renamed the two latches, but declared them at the start - hopefully that makes more sense?

Reply:
I find it much clearer, yes, thanks. This one (`roomUpdated`) is no longer mutated, so it can be a `val`.